File: NyanDroid/src/main/java/com/powerje/nyan/NyanActivity.java
package com.powerje.nyan;

import android.app.Activity;
import android.content.Intent;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.Window;

import java.io.IOException;

public class NyanActivity extends Activity {
	private NyanView mRoot = null;
	private MediaPlayer mPlayer = null;

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

        if (android.os.Build.VERSION.SDK_INT >= 11) {
            requestWindowFeature(Window.FEATURE_ACTION_BAR_OVERLAY);
            getActionBar().setBackgroundDrawable(null);
        }
	}

	public void onResume() {
		super.onResume();
        // scale by whichever is larger, width or height
        DisplayMetrics metrics = this.getResources().getDisplayMetrics();
        int h = metrics.heightPixels;
        int w = metrics.widthPixels;
        int largest = h > w ? h : w;

        mRoot = new NyanView(this, largest);
        setContentView(mRoot);

        if (mPlayer == null) {
		    mPlayer = MediaPlayer.create(this, R.raw.dyan_loop);
		    mPlayer.setLooping(true);
        } else {
            mPlayer.start();
        }

		mPlayer.setScreenOnWhilePlaying(true);

		try {
			mPlayer.prepare();
		} catch (IllegalStateException e) {
			e.printStackTrace();
		} catch (IOException e) {
			e.printStackTrace();
		}
		mPlayer.setOnPreparedListener(new MediaPlayer.OnPreparedListener() {
			public void onPrepared(MediaPlayer mp) {
				mPlayer.start();
			}
		});
	}

	public boolean onOptionsItemSelected(MenuItem item) {
		switch (item.getItemId()) {
		case R.id.menu_settings:
			startActivity(new Intent(NyanActivity.this, NyanSettings.class));
			break;
		}
		return true;
	}

	public boolean onCreateOptionsMenu(Menu menu) {
		MenuInflater inflater = getMenuInflater();
		inflater.inflate(R.menu.menu, menu);
		return true;
	}

	@Override
	public void onPause() {
		super.onPause();
		mPlayer.pause();
	}

    public void onStop() {
        super.onStop();
        mRoot.cancel();
        mRoot = null;
        System.gc();
    }

	@Override
	public void onDestroy() {
		super.onDestroy();
		mRoot = null;
	}

}

File: NyanDroid/src/main/java/com/powerje/nyan/NyanActivity.kt
package com.powerje.nyan

import android.content.Intent
import android.graphics.Color
import android.media.MediaPlayer
import android.os.Build
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.view.WindowInsets
import android.view.WindowInsetsController
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.io.IOException


class NyanActivity : AppCompatActivity() {
    private var nyanView: NyanView? = null
    private var player: MediaPlayer? = null
    private var toolbar: Toolbar? = null

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.nyan_activity)
        toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayShowTitleEnabled(false)
        window.setDecorFitsSystemWindows(false)
        val controller = window.insetsController
        if (controller != null) {
            controller.hide(WindowInsets.Type.statusBars() or WindowInsets.Type.navigationBars())
            controller.systemBarsBehavior = WindowInsetsController.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        }
        window.statusBarColor = Color.TRANSPARENT
        window.navigationBarColor = Color.TRANSPARENT

        ViewCompat.setOnApplyWindowInsetsListener(toolbar!!) { v, insets ->
            val barsInsets = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            val topBarHeight = barsInsets.top
            val layoutParams = v.layoutParams as ViewGroup.MarginLayoutParams
            layoutParams.topMargin = topBarHeight
            v.layoutParams = layoutParams

            WindowInsetsCompat.CONSUMED
        }
    }

    public override fun onResume() {
        super.onResume()

        if (player == null) {
            player = MediaPlayer.create(this, R.raw.dyan_loop)
            player!!.isLooping = true
        } else {
            player!!.start()
        }

        player!!.setScreenOnWhilePlaying(true)

        try {
            player!!.prepare()
        } catch (e: IllegalStateException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        }

        player?.run {
            setOnPreparedListener {
                it.start()
            }
        }

        nyanView = findViewById(R.id.nyan_view)
        nyanView!!.start()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.menu_settings -> startActivity(Intent(this@NyanActivity, NyanSettingsActivity::class.java))
        }
        return true
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return true
    }

    public override fun onPause() {
        super.onPause()
        player?.pause()
        nyanView?.cancel()
        nyanView = null
    }

    public override fun onStop() {
        super.onStop()
        nyanView?.cancel()
        nyanView = null
        System.gc()
    }

    public override fun onDestroy() {
        super.onDestroy()
        nyanView = null
    }
}